New GHA CI seems to not work

From the build for 674f73da63e127b143adbb613197f5630d68105d:

```
2021-03-10 17:07:14 [INFO] (mdbook::book): Book building has started
2021-03-10 17:07:14 [INFO] (mdbook::book): Running the html backend
2021-03-10 17:07:16 [INFO] (mdbook::book): Running the linkcheck backend
2021-03-10 17:07:16 [INFO] (mdbook::renderer): Invoking the "linkcheck" renderer
2021-03-10 17:07:16 [WARN] (mdbook::renderer): Renderer command `../../ci/linkcheck.sh` uses a path relative to the renderer output directory `/home/runner/work/rustc-dev-guide/rustc-dev-guide/book/linkcheck`. This was previously accepted, but has been deprecated. Relative executable paths should be relative to the book root.
error: unexpected state: BASE_SHA must be non-empty in CI
2021-03-10 17:07:16 [WARN] (mdbook::renderer): Error writing the RenderContext to the backend, Broken pipe (os error 32)
Error: -10 17:07:16 [ERROR] (mdbook::renderer): Renderer exited with non-zero return code.
Error: -10 17:07:16 [ERROR] (mdbook::utils): Error: Rendering failed
Error: -10 17:07:16 [ERROR] (mdbook::utils): 	Caused By: The "linkcheck" renderer failed
Error: Process completed with exit code 101.
```

cc @JohnTitor

--- a/ci/linkcheck.sh
+++ b/ci/linkcheck.sh
@@ -9,7 +9,7 @@ if [ "$GITHUB_EVENT_NAME" = "schedule" ] ; then # running in scheduled job
 
   echo "Doing full link check."
   set -x
-elif [ "$CI" = "true" ] ; then # running in PR CI build
+elif [ "$GITHUB_EVENT_NAME" = "pull_request" ] ; then # running in PR CI build
   if [ -z "$BASE_SHA" ]; then
     echo "error: unexpected state: BASE_SHA must be non-empty in CI"
     exit 1

